tests/lib/smf/src/test_lib_self_transition_smf.c

```diff
@@ -118,14 +150,52 @@ static struct test_object {
 	uint32_t first_time;
 } test_obj;
 
-static void parent_ab_entry(void *obj)
+static void root_entry(void *obj)
 {
 	struct test_object *o = TEST_OBJECT(obj);
 
 	o->tv_idx = 0;
 
-	zassert_equal(o->transition_bits, test_value[o->tv_idx],
-		      "Test Parent AB entry failed");
+	zassert_equal(o->transition_bits, test_value[o->tv_idx], "Test Root entry failed");
+
+	if (o->terminate == ROOT_ENTRY) {
+		smf_set_terminate(obj, -1);
+		return;
+	}
```

Comment:
This condition never occurs and isn't verified by the test.

Reply:
Non-blocking - prefer that you delete since it isn't used.

```suggestion
```

Reply:
Would it be better to build a test that tests termination in root_entry? I think the other tests have that covered with PARENT_ENTRY not wrapped in a root state.